### PR TITLE
Replace QMediaPlayer with QSoundEffect for consistent highlight audio

### PR DESCRIPTION
**Pull request by Greenlandicsmiley**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description
### Reason
Currently Chatterino2 uses the QMediaPlayer class for playing the highlight sound.
There are a two problems:

- After a few highlights the audio degrades and eventually becomes basically muted, unless the media/source is reset.
- Sometimes the highlights queue up in the background and then randomly replays the highlights, unless the media/source is reset.

And as far as I know, they are not fixed in QT6.

I've tried to find a few ways, and searched for solutions for both, but there are no solutions so far, unless we use QSoundEffect. It's a problem with the way QMediaPlayer works.

### Solution
This Pull Request should (or is an attempt to) fix the audio issues that plague Chatterino2, the highlight audio playback should be more consistent, and much like the current solution does not leak memory. It's also compatible with QT6 for the eventual migration.

### CODE REVIEW

File: src/controllers/notifications/NotificationController.cpp [hunk 1]
```diff
@@ -97,7 +97,7 @@ void NotificationController::removeChannelNotification(
 }
 void NotificationController::playSound()
 {
-    static auto player = new QMediaPlayer;
+    static QSoundEffect *player = new QSoundEffect;
```
[the remark this answers is not in the document]
Comment (Greenlandicsmiley) [reply src/controllers/notifications/NotificationController.cpp]: Should be fixed in commit e0d9030, sorry chatterino gods for not using clang :pray:

File: src/controllers/notifications/NotificationController.cpp [hunk 2]
```diff
@@ -97,7 +97,7 @@
...
```
[the remark this answers is not in the document]
Comment (Greenlandicsmiley) [reply src/controllers/notifications/NotificationController.cpp]: Should be fixed in commit e0d9030, sorry chatterino gods for not using clang :pray:

File: src/messages/SharedMessageBuilder.cpp [hunk 3]
```diff
@@ -198,11 +198,11 @@ void SharedMessageBuilder::appendChannelName()
         ->setLink(link);
 }
 
-inline QMediaPlayer *getPlayer()
+inline QSoundEffect *getPlayer()
 {
     if (isGuiThread())
     {
-        static auto player = new QMediaPlayer;
+        static QSoundEffect *player = new QSoundEffect;
```
[the remark this answers is not in the document]
Comment (Greenlandicsmiley) [reply src/messages/SharedMessageBuilder.cpp]: Should be fixed in commit e0d9030, sorry chatterino gods for not using clang :pray:

File: src/messages/SharedMessageBuilder.cpp [hunk 4]
```diff
@@ -198,11 +198,11 @@
...
```
[the remark this answers is not in the document]
Comment (Greenlandicsmiley) [reply src/messages/SharedMessageBuilder.cpp]: Should be fixed in commit e0d9030, sorry chatterino gods for not using clang :pray:

**Review by pajlada — APPROVED** [review]
On Arch Linux with Pulseaudio, this works as expected (and seems to be a bit faster than my old build. Only downside I can find is it shows up without a proper name
Old behaviour: 
image

New behaviour: 
image

On Windows 10, it works as expected and shows up in the volume mixer as expected (thanks Felanbird)
image

I'm happy to merge this in as-is, but I'll let it brew over the weekend in case anyone else has any unique setups they want to try this on.

Thank you for looking into this and finding this!

**Comment by cWve** [comment]
Tried this on Arch Linux with pipewire, seems to work well, I tried 10 pings in a row.
I also get the QtPulseAudio name on pavucontrol

**Comment by Greenlandicsmiley** [comment]
If this article is anything to go by: https://wsjt-devel.narkive.com/CQ6ciLFB/qtpulseaudio-and-pavucontrol

Volume/sound card settings will be lost everytime the program starts, because with these changes Chatterino will have "QtPulseAudio:x" as the name, where x is the PID of chatterino.

Maybe there is a way to choose a name that shows up in PulseAudio?

**Comment by Felanbird** [comment]
> Maybe there is a way to choose a name that shows up in PulseAudio?

I believe pajlada tried setting a name to no avail 
`15:33 pajlada: forsenPossessed QSoundEffect doesn't allow naming of the playback stream`

**Comment by Greenlandicsmiley** [comment]
> I believe pajlada tried setting a name to no avail

In that case, should we assume that only a small portion of the Chatterino audience care about their volume/sound card settings, and go ahead with the change?
Of course after we have confirmed the changes to be good on most if not all Linux distributions.

**Comment by pajlada** [comment]
Seems like Qt is not very good with audio atm, marking this as a draft as we explore other options

**Comment by Greenlandicsmiley** [comment]
Sound backend for Chatterino2 is being changed to miniaudio, so I am closing this PR.

**Pull request closed without merging**